modules/translation/translation.go

```diff
@@ -99,3 +100,66 @@ func (l *locale) Language() string {
 func (l *locale) Tr(format string, args ...interface{}) string {
 	return i18n.Tr(l.Lang, format, args...)
 }
+
+// Language specific rules for translating plural texts
+var trNLangRules = map[string]func(int64) int{
+	"en-US": func(cnt int64) int {
+		if cnt == 1 {
+			return 0
+		}
+		return 1
+	},
+	"lv-LV": func(cnt int64) int {
+		if cnt%10 == 1 && cnt%100 != 11 {
+			return 0
+		}
+		return 1
+	},
+	"ru-RU": func(cnt int64) int {
+		if cnt%10 == 1 && cnt%100 != 11 {
+			return 0
+		}
+		return 1
+	},
+	"zh-CN": func(cnt int64) int {
+		return 0
+	},
+	"zh-HK": func(cnt int64) int {
+		return 0
+	},
+	"zh-TW": func(cnt int64) int {
+		return 0
+	},
+	"fr-FR": func(cnt int64) int {
+		if cnt > -2 && cnt < 2 {
+			return 0
+		}
+		return 1
+	},
+}
+
+// TrN returns translated message for plural text translation
+func (l *locale) TrN(cnt interface{}, key1, keyN string, args ...interface{}) string {
+	var c int64
+	if t, ok := cnt.(int); ok {
+		c = int64(t)
+	} else if t, ok := cnt.(int16); ok {
+		c = int64(t)
+	} else if t, ok := cnt.(int32); ok {
+		c = int64(t)
+	} else if t, ok := cnt.(int64); ok {
+		c = t
+	} else {
+		return l.Tr(keyN, args...)
+	}
```

Comment:
```suggestion
	switch t := cnt.(type) {
	case int:
		c = int64(t)
	case int8:
		c = int64(t)
	case int16:
		c = int64(t)
	case int32:
		c = int64(t)
	case int64:
		c = t
	default:
		return l.Tr(keyN, args...)
	}
```

Reply:
```suggestion
	switch t := cnt.(type) {
	case int, int8, int16, int32, int64:
		c = t
	default:
		return l.Tr(keyN, args...)
	}
```

Reply:
This won't work. go(without proper generics implementation) will now mark `t` on line 146 as `interface{}`(which cannot be converted to `int64`) because go doesn't have support for 1 variable to be multiple types.

Reply:
uff 😅